cfme/containers/provider/openshift.py

```diff
@@ -113,6 +113,11 @@ def __init__(
 
     def create(self, **kwargs):
 
+        # Enable C&U for any provider with metrics provider
+        if self.metrics_type != "Disabled":
```

Comment:
Is there a reason for automatically enabling these roles, and not just enabling them when they're needed for test?

Reply:
I changed to WIP for now

Reply:
@mshriver Basically, this three roles activating the metrics collection for openshift provider (and maybe for other providers as well), so adding a metrics endpoint without enabling is not relevant at all. Since we don't have any automated test for metrics I testing these features manually and adding the provider using the automation.

Reply:
Honestly, I don't like it. If you remove openshift provider these roles will be active still.

Reply:
I agree @quarckster. I'm opposed to this change, these roles should only be enabled when needed for testing, and disabled at the end of a test (fixture setup+teardown).

@shalomnaim1 If you need these roles enabled for your manual testing, having them enabled when the provider is configured via automation is not the right solution.  You can use `miq shell` and these methods to configure the roles for yourself when testing.

This just needlessly increases the appliance worker load on all tests using this provider.

Reply:
@mshriver As long as the metrics endpoint exists, this role has to be enabled.

Reply:
@shalomnaim1 Do the tests fail without that role enabled?

Reply:
@mfalesni sure, without this roles no metrics test can't be run since CFME will not collect any metrics. In addition, if someone adding an openshift provider for some reason without enabling this role, no metrics will be collected

Reply:
That's the expected behavior with all providers, and fixtures within the tests should enable the roles on the appliance.  There are, as far as I know, no container metrics tests in automation yet. When added, they should enable the roles in a fixture as well.

Reply:
IMO, when someone adds a metrics endpoint he wants metrics to be collected, so it has to be a part of the setup and not part of the test. In the same way, you can say that if I want to test metrics on OCP I will have to create a fixture of setting up the metric endpoint, for me it just not making any sense.

Reply:
@shalomnaim1 How do you propose to disable roles when an openshift provider was removed?

Reply:
@quarckster I think it can be done by adding code which disables these roles while removing OCP provider with configured metrics endpoint or just metrics endpoint removal.